authorizerd_test.go

```diff
@@ -927,18 +928,110 @@ func Test_authorizer_verify(t *testing.T) {
 		res  string
 		cert *x509.Certificate
 	}
-	tests := []struct {
+	type test struct {
 		name       string
 		fields     fields
 		args       args
 		wantErr    bool
 		wantResult Principal
-	}{
-		// TODO: Add test cases.
+		checkFunc  func(prov *authority) error
+	}
+	tests := []test{
+		func() test {
+			c := gache.New()
+			var count int
+			pdm := &PolicydMock{
+				CheckPolicyFunc: func(ctx context.Context, domain string, roles []string, action, resource string) error {
+					count++
+					return nil
+				},
+			}
+			rt := &role.Token{}
+			p := &principal{
+				name:       rt.Principal,
+				roles:      rt.Roles,
+				domain:     rt.Domain,
+				issueTime:  rt.TimeStamp.Unix(),
+				expiryTime: rt.ExpiryTime.Unix(),
+			}
+			rpm := &RoleProcessorMock{
+				rt:      rt,
+				wantErr: nil,
+			}
+			return test{
+				name: "test disablePolicyd true",
+				fields: fields{
+					cache:          c,
+					policyd:        pdm,
+					disablePolicyd: true,
+					roleProcessor:  rpm,
+				},
+				args: args{
+					m:   roleToken,
+					ctx: context.Background(),
+					tok: "dummyTok",
+					act: "dummyAct",
+					res: "dummyRes",
+				},
+				wantErr:    false,
+				wantResult: p,
+				checkFunc: func(prov *authority) error {
+					if count != 0 {
+						return errors.New("CheckPolicy must not be called")
+					}
+					return nil
+				},
+			}
+		}(),
+		func() test {
+			c := gache.New()
+			pdm := &PolicydMock{
+				CheckPolicyFunc: func(ctx context.Context, domain string, roles []string, action, resource string) error {
+					return nil
+				},
+			}
+			rt := &role.Token{}
+			p := &principal{
+				name:       rt.Principal,
+				roles:      rt.Roles,
+				domain:     rt.Domain,
+				issueTime:  rt.TimeStamp.Unix(),
+				expiryTime: rt.ExpiryTime.Unix(),
+			}
+			rpm := &RoleProcessorMock{
+				rt:      rt,
+				wantErr: nil,
+			}
+			return test{
+				name: "test correct cache key",
+				fields: fields{
+					cache:          c,
+					policyd:        pdm,
+					disablePolicyd: true,
+					roleProcessor:  rpm,
+				},
+				args: args{
+					m:   roleToken,
+					ctx: context.Background(),
+					tok: "dummyTok",
+					act: "dummyAct",
+					res: "dummyRes",
+				},
+				wantErr:    false,
+				wantResult: p,
+				checkFunc: func(prov *authority) error {
+					_, ok := prov.cache.Get("dummyTok")
+					if !ok {
+						return errors.New("cannot get dummyTok from cache")
+					}
+					return nil
+				},
+			}
+		}(),
```

Comment:
How about adding a test case like the following?
- disableA: true
- wantResult: p
- checkFunc: check `prov.cache.Get("dummyTok" + "dummyAct" + "dummyRes")` is ok or not

Reply:
Fixed
https://github.com/yahoojapan/athenz-authorizer/pull/74/commits/734d5b538145de352454bb968c8a672635f69e5f

Reply:
Sorry, I made a mistake.
What I meant to say was the test with disablePolicyd set to false.

Fixed the comment above.

Reply:
Fixed
https://github.com/yahoojapan/athenz-authorizer/pull/74/commits/f8f3c1259d2eb446f10255846e6c2162f84b2583